homeassistant/components/pi_hole/config_flow.py

```diff
@@ -61,10 +61,11 @@ async def async_step_init(self, user_input, is_import=False):
             location = user_input[CONF_LOCATION]
             tls = user_input[CONF_SSL]
             verify_tls = user_input[CONF_VERIFY_SSL]
-            endpoint = f"{host}/{location}"
 
-            if await self._async_endpoint_existed(endpoint):
-                return self.async_abort(reason="already_configured")
+            await self.async_set_unique_id(
+                f"{user_input[CONF_HOST]}/{user_input[CONF_LOCATION]}"
+            )
+            self._abort_if_unique_id_configured()
```

Comment:
Must we still include tests for already configured integrations if that integration is using this method?

Reply:
Yes, as you need to test the unique ID is in the resulting entry.

Reply:
I tested what you've said in my test environment, removed all unique_id code and attempt to set up pihole twice. HA says already configured. Local tests still pass with removed unique ids as well. Perhaps I'm not testing properly?

Reply:
You need to adjust the unit tests testing this configuration flow, to ensure the unique id is set as expected for this configuration flow... The current tests for this integration do not cover that.

Additionally, existing entries do not have an unique_id. So if someone, today, is using this integration the unique ID in their configuration entry will be empty. It needs to be filled in order for `_abort_if_unique_id_configured` to work correctly (as it checks currently present entries in the system by their unique ID, which wasn't set before).